b2sdk/session.py

```diff
@@ -30,13 +30,22 @@ def __getattr__(self, name):
         @functools.wraps(f)
         def wrapper(*args, **kwargs):
             auth_failure_encountered = False
+            # A *magic* that will identify and generate the correct type of Url and token based on the decorator on the B2RawApi method.
+            token_type = getattr(f, 'token_type', TokenType.API)
```

Comment:
I think I might put the knowledge of "what is the default token type" in the raw API layer, next to the token type decorator.  Maybe add a `get_token_type` function there that does this `getattr`.

Reply:
great idea, didn't think of that!